meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -35,7 +42,101 @@ export const ShowStyleVariantsSettings = withTranslation()(
 
 			this.state = {
 				editedMappings: [],
+				uploadFileKey: Date.now(),
+			}
+		}
+		importVariants(e: React.ChangeEvent<HTMLInputElement>) {
+			const { t } = this.props
+
+			const file = e.target.files ? e.target.files[0] : null
+			if (!file) {
+				return
+			}
+
+			const reader = new FileReader()
+			reader.onload = (e2) => {
+				this.setState({
+					uploadFileKey: Date.now(),
+				})
+
+				const uploadFileContents = (e2.target as any).result
+
+				let newVariants: Array<ShowStyleVariant> = []
+				try {
+					newVariants = JSON.parse(uploadFileContents)
+					if (!_.isArray(newVariants)) {
+						throw new Error('Imported file did not contain an array')
+					}
+				} catch (error) {
+					NotificationCenter.push(
+						new Notification(
+							undefined,
+							NoticeLevel.WARNING,
+							t('Failed to import new showstyle variants: {{errorMessage}}', { errorMessage: error + '' }),
+							'VariantSettings'
+						)
+					)
+					return
+				}
+
+				_.forEach(newVariants, (entry) => {
+					MeteorCall.showstyles
+						.insertShowStyleVariantWithBlueprint(
+							entry.showStyleBaseId,
+							entry.blueprintConfig,
+							entry._id,
+							entry._rundownVersionHash,
+							entry.name
+						)
+						.catch((error) => {
+							NotificationCenter.push(
+								new Notification(
+									undefined,
+									NoticeLevel.WARNING,
+									t('Failed to import Variant ' + entry.name + '. Make sure it is not already imported.', {
+										errorMessage: error + '',
+									}),
+									'VariantSettings'
+								)
+							)
+						})
+				})
 			}
+			reader.readAsText(file)
+		}
+		copyVariant = (showStyleVariant: ShowStyleVariant) => {
+			MeteorCall.showstyles
+				.insertShowStyleVariantWithBlueprint(
+					showStyleVariant.showStyleBaseId,
+					showStyleVariant.blueprintConfig,
+					getRandomId(),
+					'',
+					'Copy of ' + showStyleVariant.name
+				)
+				.catch(console.error)
+		}
+		downloadVariant = (showstyleVariant: ShowStyleVariant) => {
+			const variantArray = [showstyleVariant]
+			const jsonStr = JSON.stringify(variantArray)
+
+			const element = document.createElement('a')
+			element.href = URL.createObjectURL(new Blob([jsonStr], { type: 'application/json' }))
+			element.download = `${showstyleVariant.name}_showstyleVariant_${showstyleVariant._id}.json`
+
+			document.body.appendChild(element) // Required for this to work in FireFox
+			element.click()
+			document.body.removeChild(element) // Required for this to work in FireFox
+		}
+		downloadAllVariants = () => {
```

Comment:
This and `downloadVariant` is almost identical. You should restract the download logic into it's own method that these two uses instead of duplicating the logic

Reply:
Added it to [f74f966](https://github.com/tv2/tv-automation-server-core/pull/188/commits/f74f9665479993af5c8647b78f40a11fe9370e0f).